src/test/java/eu/europeana/metis/sandbox/service/dataset/AsyncDatasetPublishServiceImplTest.java

```diff
@@ -271,30 +249,92 @@ void runHarvestOaiAsync_withXsltReachMaxRecords_expectSuccess() throws Harvester
-    when(harvestService.harvestOaiRecordHeader(anyString() ,any(OaiHarvestData.class), any(Record.RecordBuilder.class))).thenReturn(new RecordInfo(recordData));
+    when(harvestService.harvestOaiRecordHeader(anyString(), any(OaiHarvestData.class),
+        any(Record.RecordBuilder.class))).thenReturn(new RecordInfo(recordData));
 
-    service.runHarvestOaiAsync("datasetName", "datasetId", Country.NETHERLANDS, Language.NL,
-            oaiHarvestData);
+    service.runHarvestOaiAsync("datasetName", "datasetId", Country.NETHERLANDS, Language.NL, oaiHarvestData);
 
     verify(datasetService, times(1)).updateNumberOfTotalRecord("datasetId", 1);
     verify(datasetService, times(1)).updateRecordsLimitExceededToTrue("datasetId");
     verify(amqpTemplate, times(1)).convertAndSend(eq("transformationEdmExternalQueue"), captor.capture());
     assertEquals(recordData, captor.getValue().getRecord());
   }
 
+  @Test
+  void runHarvestOaiAsync_withoutXsltSkipDeletedRecords_expectSuccess() throws HarvesterException {
+    service = new AsyncDatasetPublishServiceImpl(amqpTemplate, "createdQueue", "transformationEdmExternalQueue", taskExecutor,
+        oaiHarvester, harvestService, datasetService, 1);
+
+    Record recordData = Record.builder().country(Country.NETHERLANDS).language(Language.NL).datasetName("datasetName")
+        .datasetId("datasetId").content(new byte[0]).build();
+
+    OaiHarvestData oaiHarvestData = new OaiHarvestData("url", "setspec", "metadaformat", "oaiIdentifier");
+
+    OaiRecordHeader element1 = new OaiRecordHeader("oaiIdentifier1", true, Instant.now());
+    OaiRecordHeader element2 = new OaiRecordHeader("oaiIdentifier2", false, Instant.now());
+    List<OaiRecordHeader> iteratorList = new ArrayList<>();
+    iteratorList.add(element1);
+    iteratorList.add(element2);
+    OaiRecordHeaderIterator oaiRecordHeaderIterator = new TestHeaderIterator(iteratorList);
+
+    when(oaiHarvester.harvestRecordHeaders(any(OaiHarvest.class))).thenReturn(oaiRecordHeaderIterator);
+    when(datasetService.isXsltPresent(anyString())).thenReturn(false);
+    when(harvestService.harvestOaiRecordHeader(anyString(), any(OaiHarvestData.class),
+        any(Record.RecordBuilder.class))).thenReturn(new RecordInfo(recordData));
+
+    service.runHarvestOaiAsync("datasetName", "datasetId", Country.NETHERLANDS, Language.NL, oaiHarvestData);
+
+    verify(datasetService, times(1)).updateNumberOfTotalRecord("datasetId", 1);
+    verify(datasetService, times(0)).updateRecordsLimitExceededToTrue("datasetId");
+    verify(amqpTemplate, times(1)).convertAndSend(eq("createdQueue"), captor.capture());
+    assertEquals(recordData, captor.getValue().getRecord());
+  }
+
+  @Test
+  void runHarvestOaiAsync_withXsltSkipDeletedRecords_expectSuccess() throws HarvesterException {
+    service = new AsyncDatasetPublishServiceImpl(amqpTemplate, "createdQueue", "transformationEdmExternalQueue", taskExecutor,
```

Comment:
also for test initialization you can extract this into a method. I see is duplicated
   ```
 service = new AsyncDatasetPublishServiceImpl(amqpTemplate, "createdQueue", "transformationEdmExternalQueue", taskExecutor,
        oaiHarvester, harvestService, datasetService, 1);

    Record recordData = Record.builder().country(Country.NETHERLANDS).language(Language.NL).datasetName("datasetName")
        .datasetId("datasetId").content(new byte[0]).build();

    OaiHarvestData oaiHarvestData = new OaiHarvestData("url", "setspec", "metadaformat", "oaiIdentifier");

    OaiRecordHeader element1 = new OaiRecordHeader("oaiIdentifier1", true, Instant.now());
    OaiRecordHeader element2 = new OaiRecordHeader("oaiIdentifier2", false, Instant.now());
    List<OaiRecordHeader> iteratorList = new ArrayList<>();
    iteratorList.add(element1);
    iteratorList.add(element2);
    OaiRecordHeaderIterator oaiRecordHeaderIterator = new TestHeaderIterator(iteratorList);
```

Reply:
done